src/libclipper/include/clipper/containers.hpp

```diff
@@ -21,7 +21,9 @@ class ModelContainer {
  public:
   ~ModelContainer() = default;
   ModelContainer(VersionedModelId model, int container_id, int replica_id,
-                 InputType input_type);
+                  InputType input_type);
+  ModelContainer(VersionedModelId model, int container_id, int replica_id,
+                 InputType input_type, boost::optional<int> batch_size);
```

Comment:
If we're always specifying `batch_size` as either `-1` or a positive integer, we don't need to use an optional here. I had originally suggested passing an absent optional value to the `ModelContainer` constructor if the user has not defined the batch size, but using `-1` directly seems fine. Change this parameter type and the `batch_size_` class member type to `int`.